build-tools/installers/create-installers.targets

```diff
@@ -0,0 +1,122 @@
+    <MSBuildFiles Include="$(MSBuildSrcDir)\**\proguard.jar"></MSBuildFiles>
+    <MSBuildFiles Include="$(MSBuildSrcDir)\**\license.html"></MSBuildFiles>
+    <MSBuildFiles Include="$(MSBuildSrcDir)\r8.jar"></MSBuildFiles>
+    <MSBuildFiles Include="$(MSBuildSrcDir)\SgmlReaderDll.dll"></MSBuildFiles>
+    <MSBuildFiles Include="$(MSBuildSrcDir)\Xamarin*"></MSBuildFiles>
+  </ItemGroup>
+  <ItemGroup> <!-- monodroid -->
+    <MSBuildFiles Condition="Exists('$(MSBuildSrcDir)\ICSharpCode.SharpZipLib.dll')" Include="$(MSBuildSrcDir)\ICSharpCode.SharpZipLib.dll"></MSBuildFiles>
+    <MSBuildFiles Condition="Exists('$(MSBuildSrcDir)\INIFileParser.dll')" Include="$(MSBuildSrcDir)\INIFileParser.dll"></MSBuildFiles>
+    <MSBuildFiles Condition="Exists('$(MSBuildSrcDir)\jar2xml.jar')" Include="$(MSBuildSrcDir)\jar2xml.jar"></MSBuildFiles>
+    <MSBuildFiles Condition="Exists('$(MSBuildSrcDir)\Mono.AndroidTools.dll')" Include="$(MSBuildSrcDir)\Mono.AndroidTools.dll"></MSBuildFiles>
+    <MSBuildFiles Include="$(MSBuildSrcDir)\Mono.Android.DebugRuntime-*"/>
+  </ItemGroup>
+  <ItemGroup>
+    <MSBuildFilesWin Include="$(MSBuildSrcDir)\cross-arm.exe"></MSBuildFilesWin>
+    <MSBuildFilesWin Include="$(MSBuildSrcDir)\cross-arm64.exe"></MSBuildFilesWin>
+    <MSBuildFilesWin Include="$(MSBuildSrcDir)\cross-x86.exe"></MSBuildFilesWin>
+    <MSBuildFilesWin Include="$(MSBuildSrcDir)\cross-x86_64.exe"></MSBuildFilesWin>
+    <MSBuildFilesWin Include="$(MSBuildSrcDir)\opt.exe"></MSBuildFilesWin>
+    <MSBuildFilesWin Include="$(MSBuildSrcDir)\llc.exe"></MSBuildFilesWin>
+    <MSBuildFilesWin Include="$(MSBuildSrcDir)\**\proguard.bat"></MSBuildFilesWin>
+  </ItemGroup>
+  <ItemGroup>
+    <MSBuildHostFilesUnix Include="$(MSBuildSrcDir)\$(HostOS)\cross-arm"></MSBuildHostFilesUnix>
+    <MSBuildHostFilesUnix Include="$(MSBuildSrcDir)\$(HostOS)\cross-arm64"></MSBuildHostFilesUnix>
+    <MSBuildHostFilesUnix Include="$(MSBuildSrcDir)\$(HostOS)\cross-x86"></MSBuildHostFilesUnix>
+    <MSBuildHostFilesUnix Include="$(MSBuildSrcDir)\$(HostOS)\cross-x86_64"></MSBuildHostFilesUnix>
+    <MSBuildHostFilesUnix Include="$(MSBuildSrcDir)\$(HostOS)\illinkanalyzer"></MSBuildHostFilesUnix>
+    <MSBuildHostFilesUnix Include="$(MSBuildSrcDir)\$(HostOS)\mono"></MSBuildHostFilesUnix>
+    <MSBuildHostFilesUnix Include="$(MSBuildSrcDir)\$(HostOS)\mono-symbolicate"></MSBuildHostFilesUnix>
+    <MSBuildHostFilesUnix Include="$(MSBuildSrcDir)\$(HostOS)\opt"></MSBuildHostFilesUnix>
+    <MSBuildHostFilesUnix Include="$(MSBuildSrcDir)\$(HostOS)\llc"></MSBuildHostFilesUnix>
+    <MSBuildFilesUnix Include="$(MSBuildSrcDir)\**\proguard.sh"></MSBuildFilesUnix>
+    <MSBuildFilesUnix Include="$(MSBuildSrcDir)\libzip.5.0.$(LibExtension)"></MSBuildFilesUnix>
+  </ItemGroup>
+  <ItemGroup>
+    <MSBuildLibFiles Include="$(MSBuildSrcDir)\lib\**" Exclude="$(MSBuildSrcDir)\lib\**\*.d.so;$(MSBuildSrcDir)\lib\host-*\**"></MSBuildLibFiles>
```

Comment:
This wildcard will cause default inclusion of Windows files in the macOS `.pkg`, and macOS files in the Windows `.vsix` (once we migrate over `create-vsix.csproj`.

This may be a case where an intermediate item group is handy:

```xml
<ItemGroup>
  <_TargetAbi Include="arm64-v8a" />
  <_TargetAbi Include="armeabi-v7a" />
  <_TargetAbi Include="x86" />
  <_TargetAbi Include="x86_64" />
  <_HostAbi Include="host-Linux" Condition=" '$(HostOS)' == 'Linux' "/>
  <_HostAbi Include="host-Darwin" Condition=" '$(HostOS)' == 'Darwin' "/>
  <_HostAbi Include="host-mxe-Win32" Condition=" '$(HostOS)' == 'Windows' "/>
  <_HostAbi Include="host-mxe-Win64" Condition=" '$(HostOS)' == 'Windows' "/>
</ItemGroup>
<ItemGroup>
  <MSBuildLibFiles Include="@(_TargetAbi->'$(MSBuildSrcDir)\lib\%(Identity)\lib*.so" Condition="Exists('$(MSBuildSrcDir)\lib\%(Identity)'" />
  <MSBuildLibFiles Exclude="@(_TargetAbi->'$(MSBuildSrcDir)\lib\%(Identity)\lib*.d.so" Condition="Exists('$(MSBuildSrcDir)\lib\%(Identity)'" />
  <MSBuildLibFiles Include="@(_HostAbi->'$(MSBuildSrcDir)\lib\%(Identity)\*.$(LibExtension)" />
  <MSBuildLibFiles Exclude="@(_HostAbi->'$(MSBuildSrcDir)\lib\%(Identity)\*.d.$(LibExtension)" />
</ItemGroup>
```

I wonder if it would be better to explicitly list the files, though.  It would be more verbose, but that might be a feature, not a bug:

```xml
<ItemGroup Condition=" '$(HostOS)' == 'Darwin' ">
  <MSBuildLibFiles Include="$(MSBuildSrcDir)\lib\host-mxe-Win32/libmono-profiler-log.dylib" />
  <MSBuildLibFiles Include="$(MSBuildSrcDir)\lib\host-mxe-Win32/libmono-profiler-aot.dylib" />
  <MSBuildLibFiles Include="$(MSBuildSrcDir)\lib\host-mxe-Win32/libMonoPosixHelper.dylib" />
  <MSBuildLibFiles Include="$(MSBuildSrcDir)\lib\host-mxe-Win32/libmonosgen-2.0.dylib" />
  <MSBuildLibFiles Include="$(MSBuildSrcDir)\lib\host-mxe-Win32/libmono-android.debug.dylib" />
  <MSBuildLibFiles Include="$(MSBuildSrcDir)\lib\host-mxe-Win32/libmono-android.release.dylib" />
</ItemGroup>
<ItemGroup Condition=" '$(HostOS)' == 'Windows' ">
  <MSBuildLibFiles Include="$(MSBuildSrcDir)\lib\host-mxe-Win32/libmonosgen-2.0.dll" />
  <MSBuildLibFiles Include="$(MSBuildSrcDir)\lib\host-mxe-Win32/libMonoPosixHelper.dll" />
</ItemGroup>
```

Being explicit makes it easier to reason about what *should* be in the package without having a build tree hanging around, which is a useful feature.

Additionally, it demonstrates where we have *defects* with the *current* macOS installer, as it's not properly excluding the `.d.dylib` files; this exists!

    /Library/Frameworks/Xamarin.Android.framework/Versions/Current/lib/xamarin.android/xbuild/Xamarin/Android/lib/host-Darwin//libmono-android.debug.d.dylib

Reply:
Line 102 above excludes all `host-*` folders, and as a result only pulls in the "Android flavored" versions of these libraries (`armeabi-v7a`, `x86`, `arm64-v8a`, `x86_64`). The two subsequent lines then collect the host files required by the Windows or Unix installers, respectively.

I am happy to update this to list individual files we're interested in instead, I agree that we're probably better served by being more explicit here. I think we'll need to continue to use Windows or Unix specific item groups for these files however, as we currently build our .vsix installers on macOS. Therefore we can't rely on `$(HostOS)` to determine whether or not a file should be included in the Windows installer.

@jonpryor I had also made a conscious decision to drop `host-mxe-Win32` files from the Windows side at this time, let me know if that should be reconsidered. From what I gather, only `host-mxe-Win32/libMonoPosixHelper.dll` and `host-mxe-Win32/libmonosgen-2.0.dll` end up making it into the .vsix installers, and 32bit `libmono-android.*` libs don't appear to be needed by anyone (see https://github.com/xamarin/xamarin-android/issues/2574#issuecomment-461689235).